compiler/natives/src/reflect/reflect.go

```diff
@@ -832,6 +832,21 @@ func cvtSliceArrayPtr(v Value, t Type) Value {
 	return Value{t.common(), unsafe.Pointer(array.Unsafe()), v.flag&^(flagIndir|flagAddr|flagKindMask) | flag(Ptr)}
 }
 
+// convertOp: []T -> [N]T
+func cvtSliceArray(v Value, t Type) Value {
```

Comment:
Just to double-check, does the upstream also copy the data, or return a pointer to the shared array?

Reply:
So, I ran into this. I was trying to follow what Flimzy had done for `[]T -> *[N]T` which makes a shared array and [the `TestConvertSlice2Array` test](https://cs.opensource.google/go/go/+/refs/tags/go1.20.14:src/reflect/all_test.go;l=4706-4707) caught that. The test says,
> Converting a slice to non-empty array needs to return a non-addressable copy of the original memory.

So upstream makes a copy? And by making a copy here the tests pass.

It is odd that `[]T -> *[N]T` would act differently than `[]T -> [N]T` but I think it's because `*[N]T` is a pointer to a potential `[N]T` where as `[N]T` is allocated memory for the variable. So you could point `*[N]T` at another footprint but `[N]T` is a new one. Or something like that?

Reply:
Yep, I think you are correct, thanks for confirming!